broker/consumer/rubin/README.md

```diff
@@ -0,0 +1,321 @@
+
+## Download with a `GET` request, and read the alert's schema version from the Confluent Wire header
+
+In the future, we should download schemas from the Confluent Schema Registry and store them (assuming we do not use the schema registry directly in the Kafka connector).
+Then for each alert, check the schema version in the Confluent Wire header, and load the schema file using `fastavro`.
+
+Recommendation from Spencer Nelson:
+
+> You might want to look at how Rubin's alert database ingester works. It does the same steps of deserializing alert packets, but uses the schema registry instead of lsst.alert.packet:
+
+> https://github.com/lsst-dm/alert_database_ingester/blob/main/alertingest/ingester.py#L192-L209
+> https://github.com/lsst-dm/alert_database_ingester/blob/main/alertingest/schema_registry.py
+
+Pub/Sub topics can be configured with an Avro schema attached, but it cannot be changed once attached.
+We would have to create a new topic for every schema version.
+Therefore, I don't think we should do it this way.
+
+### Download a schema from the Confluent Schema Registry using a `GET` request
+
+```bash
+SCHEMA_REGISTRY_BASIC_AUTH_USER_INFO=$KAFKA_USERNAME:$KAFKA_PASSWORD
+SCHEMA_REGISTRY_URL="https://alert-schemas-int.lsst.cloud"
+schema_version=1
+fout_rubinschema="rubinschema_v${schema_version}.avsc"
+
+# get list of schema subjects
+curl --silent -X GET -u $SCHEMA_REGISTRY_BASIC_AUTH_USER_INFO $SCHEMA_REGISTRY_URL/subjects
+# download a particular schema
+curl --silent -X GET -u $SCHEMA_REGISTRY_BASIC_AUTH_USER_INFO $SCHEMA_REGISTRY_URL/schemas/ids/${schema_version} > $fout_rubinschema
+```
+
+### Read the alert's schema version from the Confluent Wire header
+
+The following is copied from
+https://github.com/lsst-dm/alert_stream/blob/main/python/lsst/alert/stream/serialization.py
+
+```python
+import struct
+
+_ConfluentWireFormatHeader = struct.Struct(">bi")
+
+def deserialize_confluent_wire_header(raw):
+    """Parses the byte prefix for Confluent Wire Format-style Kafka messages.
+    Parameters
+    ----------
+    raw : `bytes`
+        The 5-byte encoded message prefix.
+    Returns
+    -------
+    schema_version : `int`
+        A version number which indicates the Confluent Schema Registry ID
+        number of the Avro schema used to encode the message that follows this
+        header.
+    """
+    _, version = _ConfluentWireFormatHeader.unpack(raw)
+    return version
+
+header_bytes = alert_bytes[:5]
+schema_version = deserialize_confluent_wire_header(header_bytes)
+```
+
+## Use the Confluent Schema Registry with the Kafka Connector
+
+Kafka Connect can use the Confluent Schema Registry directly.
+But schemas are stored under subjects and Kafka Connect is picky about how those
+subjects are named.
+See
+https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#subject-name-strategy
+**Rubin has set the schema subject name to “alert-packet”**, which does not conform
```

Comment:
What was the response from Rubin when you asked about this choice and its incompatibility making thing easy with Kafka Connect?

Reply:
Alternatively, can accept such a format be upstreamed to Kafka Connect so that it will work in the future?